server/service/validator.js

```diff
@@ -0,0 +1,66 @@
+const {BadRequestError} = require('../errorTypes/errorTypes')
+module.exports = {
+    validateAttributes: (modelAttributes, attributes) => {
+        //filtering attributes which aren't in the schema
+
+        const modelKeys = Object.keys(modelAttributes);
+        const filteredAttributes = Object.keys(attributes)
+            .filter(key => modelKeys.includes(key))
+            .reduce((acc, key) => Object.assign(acc, { [key]: attributes[key] }), {});
+
+
+        //check required
+        modelKeys.forEach(attributeKey => {
+            if (modelAttributes[attributeKey].required && filteredAttributes[attributeKey] == undefined) {
+                throw new BadRequestError(`${attributeKey} must be provided`)
+            }
+        });
+
+        //check types
+        Object.keys(filteredAttributes).forEach(key => {
+            const constructorName = filteredAttributes[key].constructor.name
+            if (constructorName.toLowerCase() !== modelAttributes[key].type) {
```

Comment:
if you wand to do that this way, then you need update all schemes to have proper value in type parameter.
for example 'string' -> 'String'

Reply:
Updated the types names and deleted toLowerCase()